### ISSUE TITLE
After/Before tests cannot work against data sets having a single time

### ISSUE THREAD

**Issue by aaime** [opengeospatial/ets-wfs20#75]
The current test implementation seems to work as follows:
- Collects all the values of a temporal property and builds an interval that contains them all by removing one hour from the low, adding one hour to the highest
- Picks the interval and splits it in even ranges (total duration / number of items)
- Runs before/after tests assuming at least one value should be returned

However, when the domain of the time attribute is a single value, the following happens:
- The single value T is picked and then an interval of [T-1, T+1]
- The interval split in two parts [T -1, T], [T, T+1]
- Before/After tests are run, but since the operators are not "inclusive" of boundaries, no result is returned (at least, that's what apparent from reading https://www.iso.org/obp/ui/#iso:std:iso:19108:ed-1:v1:cor:1:v1:en , using a closed spec as a reference does not help here )

The dataset might seem "odd", but it's really the WFS 1.1 test data here being used in WFS 2.0 compliance tests, here is an excerpt of the table data (dateTimeProperty is used for the tests):

````
                | intProperty |          uriProperty           | measurand |    dateTimeProperty    |      dateProperty      | decimalProperty |  id  
----------------+-------------+--------------------------------+-----------+------------------------+------------------------+-----------------+------
                |         155 | http://www.opengeospatial.org/ |     12765 |                        | 2006-10-25 02:00:00+02 |            5.03 | f001
                |         154 | http://www.opengeospatial.org/ |     12769 |                        | 2006-10-23 02:00:00+02 |            4.02 | f002
BDF4F8D978E4740 |         180 |                                |     672.1 |                        | 2006-09-01 02:00:00+02 |           12.92 | f003
                |         300 |                                |     783.5 | 2006-06-28 07:08:00+02 | 2006-12-12 01:00:00+01 |           18.92 | f008
                |        -900 |                                |       2.4 |                        |                        |             7.9 | f015
````

**Comment by lgoltz** [comment]
Thank you for reporting. 
@aaime Can you provide a service for testing?

The tests AfterTests#afterPeriod,  AfterTests#afterInstant, AfterTests#afterInstantWithOffset, and BeforeTests#beforePeriod may be affected.

Interval for the tests  must be changed to e.g. [T -2, T-1], [T+1, T+2].

Relevant parts from https://www.iso.org/obp/ui/#iso:std:iso:19108:ed-1:v1:cor:1:v1:en:
```
b) If this TM_Primitive is a TM_Period and other is a TM_Instant, the operation shall return a value for TM_RelativePosition as follows:

Before | self.end.position < other.positionition
After | self.begin.position > other.position

d) If both this TM_Primitive and other are TM_Periods, the operation shall return a value for TM_RelativePosition as follows:

Before | self.end.position < other.begin.position
After | self.begin.position > other.end.position
```

**Comment by aaime** [comment]
@lgoltz I don't have a service to test this handy, I'll check internally if we can stand up one (or add the relevant layers to an existing one)

**Comment by aaime** [comment]
Never mind, we actually have one :-)
This capabilities document contains only a single feature type, with the dateTimeProperty having a single value:

http://cloudsdi.geo-solutions.it/geoserver/sf/PrimitiveGeoFeature/ows?service=WFS&version=2.0.0&request=GetCapabilities

**Comment by lgoltz** [comment]
Fixed with #87. The created interval was splitted in three parts to ensure that the requested date/time is  useable for single date/time property values. Furthermore skipping of the test was removed and the created interval increased by one day at the end and beginning.
